gtsfm/evaluation/compare_metrics.py

```diff
@@ -0,0 +1,76 @@
+"""Function supporting metric comparisons across SfM pipelines (GTSfM, COLMAP, etc.)
+
+This function converts outputs from SfM pipelines into a format matching
+metrics from GTSfM.
+
+Authors: Jon Womack
+"""
+import os
+from typing import Dict, List
+
+import gtsfm.utils.io as io_utils
+from gtsfm.evaluation.metrics import GtsfmMetric, GtsfmMetricsGroup
+
+import thirdparty.colmap.scripts.python.read_write_model as colmap_io
+
+
+def compare_metrics(
+    txt_metric_paths: Dict[str, str],
+    json_path: str,
+    metric_filenames: List[str],
+) -> None:
+    """Converts the outputs of other SfM pipelines to GTSfMMetricsGroups saved as json files.
+
+    Creates folders for each additional SfM pipeline that contain GTSfMMetricsGroups
+    containing the same metrics as GTSFM_MODULE_METRICS_FNAMES. If one of the GTSfM metrics
+    is not available from another SfM pipeline, then the metric is left blank for that pipeline.
+
+    Args:
+        txt_metric_paths: a list of paths to directories containing outputs of other SfM pipelines
+          in COLMAP format i.e. cameras.txt, images.txt, and points3D.txt files.
+        json_path: Path to folder that contains metrics as json files.
+        metric_filenames: List of GTSfM metrics filenames.
+    """
+    for pipeline_name in txt_metric_paths.keys():
+        cameras, images, points3d = colmap_io.read_model(path=txt_metric_paths[pipeline_name], ext=".txt")
+        cameras, images, image_files, sfmtracks = io_utils.colmap2gtsfm(cameras, images, points3d, load_sfmtracks=True)
+        num_cameras = len(cameras)
+        track_lengths = []
+        image_id_num_measurements = {}
+        for track in sfmtracks:
+            track_lengths.append(track.number_measurements())
+            for k in range(track.number_measurements()):
+                image_id, uv_measured = track.measurement(k)
+                if image_id not in image_id_num_measurements:
+                    image_id_num_measurements[image_id] = 1
+                else:
+                    image_id_num_measurements[image_id] += 1
+
+        colmap_metrics = {
+            "number_cameras": GtsfmMetric("number_cameras", num_cameras),
+            "3d_tracks_length": GtsfmMetric(
+                "3d_tracks_length",
+                track_lengths,
+                plot_type=GtsfmMetric.PlotType.HISTOGRAM,
+            ),
+        }
```

Comment:
could you separate this logic where we actually compute the metrics from other pipeline data to a different function? This would help add more metrics later if we need to.

Reply:
Good point. Separated.